cockatrice/src/pictureloader.cpp

```diff
@@ -114,8 +115,23 @@ PictureLoaderWorker::PictureLoaderWorker()
     connect(this, SIGNAL(startLoadQueue()), this, SLOT(processLoadQueue()), Qt::QueuedConnection);
     connect(&SettingsCache::instance(), SIGNAL(picsPathChanged()), this, SLOT(picsPathChanged()));
     connect(&SettingsCache::instance(), SIGNAL(picDownloadChanged()), this, SLOT(picDownloadChanged()));
+    connect(&SettingsCache::instance(), &SettingsCache::networkCacheSizeChanged, this,
+            &PictureLoaderWorker::networkCacheSizeChanged);
 
     networkManager = new QNetworkAccessManager(this);
+    // We need a timeout to ensure requests don't hang indefinitely in case of
+    // cache corruption, see related Qt bug: https://bugreports.qt.io/browse/QTBUG-111397
+    // Use Qt's default timeout (30s, as of 2023-02-22)
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+    networkManager->setTransferTimeout();
+#endif
+    auto cache = new QNetworkDiskCache(this);
```

Comment:
why not assign the cache as a member?

Reply:
Didn't seem necessary, it's accessible through `networkManager->cache()` already. Do you want me to make it a member?

Reply:
it just seems simpler than using `networkManager->cache()` to me? it's just a pointer

Reply:
the networkmanager does take ownership of the cache, so we could be left with a dangling pointer, in practice we assign the network manager as a child here and never delete it otherwise so I'd say that isn't a concern either.

Reply:
I got rid of the setNetworkCacheSize on the worker and connected the settings to `QNetworkDiskCache::setMaximumCacheSize` directly so we don't need it outside of the constructor now, seems cleaner anyways